runtime/lua/vim/filetype/detect.lua

```diff
@@ -1,8 +1,18 @@
 local M = {}
 
+---@private
+local function count_lines(bufnr)
+  return vim.api.nvim_buf_line_count(bufnr)
+end
+
+---@private
+local function getline(bufnr, lnum)
+  return vim.api.nvim_buf_get_lines(bufnr, lnum-1, lnum, false)[1] or ""
+end
```

Comment:
Use the new implementation of this from here: https://github.com/neovim/neovim/pull/18219/files#diff-ddde6eb58e360a5e40cec2d0c3e175a5d2fb694fb51a33fa897e806d0c7c98c0R23-R26